Hints.py

```diff
@@ -322,6 +323,22 @@ def get_woth_hint(spoiler, world, checked):
         and location.item.name not in world.item_hint_type_overrides['woth'],
         locations))
 
+    # Choose a opportunity hint instead if either there are no woth remaining or randomly if there aren't many woth remaining
+    if not locations or (len(locations) < 3 and random.random() > 0.5):
+        opportunity_locations = spoiler.opportunity_locations[world.id]
+        opportunity_locations = list(filter(lambda location:
+            location.name not in checked
+            and not (world.woth_dungeon >= world.hint_dist_user['dungeons_woth_limit'] and location.parent_region.dungeon)
+            and location.name not in world.hint_exclusions
+            and location.name not in world.hint_type_overrides['woth']
+            and location.item.name not in world.item_hint_type_overrides['woth'],
+            opportunity_locations))
+
+        # only select an from the opportunity locations if there are any
+        if opportunity_locations:
+            hint_suffix = "contains an opportunity."
```

Comment:
```suggestion
            hint_suffix = "presents an opportunity."
```

Maybe this would sound a bit less dry?